diktat-maven-plugin/src/main/kotlin/org/cqfn/diktat/plugin/maven/DiktatBaseMojo.kt

```diff
@@ -0,0 +1,130 @@
+     */
+    abstract fun runAction(params: KtLint.Params)
+
+    /**
+     * Perform code check using diktat ruleset
+     *
+     * @throws MojoFailureException if code style check was not passed
+     * @throws MojoExecutionException if [RuleExecutionException] has been thrown
+     */
+    override fun execute() {
+        val configFile = resolveConfig()
+        if (!File(configFile).exists()) {
+            throw MojoExecutionException("Configuration file $configFile doesn't exist")
+        }
+        log.info("Running diKTat plugin with configuration file $configFile and inputs $inputs")
+
+        val ruleSets by lazy {
+            listOf(DiktatRuleSetProvider(configFile).get())
+        }
+        val lintErrors = mutableListOf<LintError>()
+
+        inputs
+            .map(::File)
+            .forEach {
+                checkDirectory(it, lintErrors, ruleSets)
+            }
+
+        reporter.afterAll()
+        if (lintErrors.isNotEmpty()) {
+            throw MojoFailureException("There are ${lintErrors.size} lint errors")
+        }
+    }
+
+    private fun resolveConfig(): String {
+        if (File(diktatConfigFile).isAbsolute) {
+            return diktatConfigFile
+        }
+
+        return generateSequence(mavenProject) { it.parent }
+            .map { File(it.basedir, diktatConfigFile) }
+            .first { it.exists() }
+            .absolutePath
+    }
+
+    /**
+     * @throws MojoExecutionException if [RuleExecutionException] has been thrown by ktlint
+     */
+    private fun checkDirectory(directory: File, lintErrors: MutableList<LintError>, ruleSets: Iterable<RuleSet>) {
+        directory
+            .walk()
+            .filter { file ->
+                file.isDirectory || file.extension.let { it == "kt" || it == "kts" }
+            }
+            .filter { it.isFile }
+            .forEach { file ->
+                log.info("Checking file $file")
+                val text = file.readText()
+                try {
+                    reporter.before(file.path)
+                    val params =
+                        KtLint.Params(
+                                fileName = file.name,
+                                text = text,
+                                ruleSets = ruleSets,
+                                userData = mapOf("file_path" to file.path),
+                                script = file.extension.equals("kts", ignoreCase = true),
+                                cb = { lintError, isCorrected ->
+                                    reporter.onLintError(file.path, lintError, isCorrected)
+                                    lintErrors.add(lintError)
+                                },
+                                debug = debug
+                        )
+                    runAction(params)
+                    reporter.after(file.path)
+                } catch (e: RuleExecutionException) {
+                    log.error("Received exception", e)
```

Comment:
please provide more information in error

Reply:
Done